File: rabbitmq/src/main/java/com/chat/model/RabbitUserEntity.java
package com.chat.model;

import java.io.Serializable;
import java.math.BigInteger;

public record RabbitUserEntity(BigInteger id,
                               String username,
                               String name,
                               String role,
                               String status,
                               String password,
                               Long createdAt,
                               Long deletedAt,
                               Long updatedAt
) implements Serializable {

}

File: rabbitmq/src/main/kotlin/com/chat/model/RabbitUserEntity.kt
package com.chat.model

import java.io.Serializable
import java.math.BigInteger


data class RabbitUserEntity(
    val id: BigInteger,
    val username: String,
    val name: String,
    val roles: String,
    val status: String,
    val password: String,
    val createdAt: Long,
    val updatedAt: Long
) : Serializable {
    constructor() : this(BigInteger.valueOf(1L), "", "", "", "", "", 1L, 1L)
}
